File: Data Structures & Algorithms/generate-parentheses/submission-0.py
class Solution:
    def generateParenthesis(self, n: int) -> List[str]:
        ans=[]
        def backtrack(cur,opencount,closedcount):
            if(len(cur)==2*n):
                ans.append(cur)
            if(opencount<n):
                backtrack(cur+"(",opencount+1,closedcount)
            if(closedcount<opencount):
                backtrack(cur+")",opencount,closedcount+1)
        backtrack("",0,0)
        return ans

File: Data Structures & Algorithms/k-closest-points-to-origin/submission-0.java
class Solution {
    public int[][] kClosest(int[][] points, int k) {
        PriorityQueue<int []> heap=new PriorityQueue<>((a,b)->(a[0]-b[0]));
        for(int [] p:points){
            int dist=p[0]*p[0]+p[1]*p[1];
            heap.add(new int[]{dist,p[0],p[1]});
        }
        int [][]ans= new int[k][2];
        for(int i=0;i<k;i++){
            int[] cur=heap.poll();
            ans[i][0]=cur[1];
            ans[i][1]=cur[2];
        }
        return ans;
    }
}
